src/controllers/automation/automationGridCtrl.js

```diff
@@ -109,6 +126,19 @@
       }
     }
 
+    function getViewByUrl(){
+      return  $window.location.href.indexOf("/selectAutomationType") > 0 ? 'TYPES': 'GRID';
+    }
+
+    window.addEventListener("popstate", function (evt) {
+      $scope.automationView = getViewByUrl();
+      $scope.$apply();
+    });
+
+    $scope.returnGrid = function() {
```

Comment:
This will return to the previous page, generally to the grid, but it could be another. What do you think about naming this function simply like "cancel" or something like this?